code-submissions-clustering-core/src/test/kotlin/org/jetbrains/research/code/submissions/clustering/LoadGraphTest.kt

```diff
@@ -54,27 +54,32 @@ class LoadGraphTest : ParametrizedBaseWithPythonSdkTest(getTmpProjectDir()) {
                 dataFrameOf("id", "step_id", "code")(
                     1, 1000, "print(1)",
                 ),
-                "([Submission(id=1, stepId=1000, code=print(1)\n)], [])"
+                "([\n" +
+                    "SubmissionsNode(code = \n" +
+                    "print(1)\n, \n" +
+                    "idList = [1])], [])"
             ),
             Arguments.of(
                 dataFrameOf("id", "step_id", "code")(
                     2, 1000, "y = 1\n",
                     3, 1000, "var = 1\n",
                     4, 1000, "a=1\n",
                 ),
-                "([Submission(id=2, stepId=1000, code=v1 = 1\n), " +
-                    "Submission(id=3, stepId=1000, code=v1 = 1\n), " +
-                    "Submission(id=4, stepId=1000, code=v1 = 1\n)], [])"
+                "([\n" +
+                    "SubmissionsNode(code = \n" +
+                    "v1 = 1\n, \n" +
+                    "idList = [2, 3, 4])], [])"
```

Comment:
Please, start ids from 1 - 1, 2, 3 (and for the next test too). But for this to work, please delete the temporary directory you created on the current test in `testLoadGraphFromDataFrame` after the test (e.g. `File(getTmpProjectDir()).deleteRecursively()`). This will avoid the problem with the same IDs for dummy

Reply:
Done! Also deleted all created virtual files as they turned out to be cached as well